src/block_ffm.rs

```diff
@@ -998,7 +1004,100 @@ B,featureB
             }],
             3,
         );
-        assert_eq!(spredict2(&mut bg, &fb, &mut pb, true), 0.5);
+        assert_eq!(spredict2(&mut bg, &fb, &mut pb, true, false), 0.5);
         assert_eq!(slearn2(&mut bg, &fb, &mut pb, true), 0.5);
     }
+
+    fn test_ffm_monte_carlo() {
+        let mut mi = model_instance::ModelInstance::new_empty().unwrap();
+        mi.learning_rate = 0.1;
+        mi.ffm_learning_rate = 0.1;
+        mi.power_t = 0.0;
+        mi.ffm_power_t = 0.0;
+        mi.bit_precision = 18;
+        mi.ffm_k = 1;
+        mi.ffm_bit_precision = 18;
+        mi.ffm_fields = vec![vec![], vec![]]; // This isn't really used
+        mi.optimizer = Optimizer::AdagradLUT;
+
+        // Nothing can be learned from a single field in FFMs
+        let mut bg = BlockGraph::new();
+        let ffm_block = new_ffm_block(&mut bg, &mi).unwrap();
+        let loss_block = block_loss_functions::new_logloss_block(&mut bg, ffm_block, true);
+        bg.finalize();
+        bg.allocate_and_init_weights(&mi);
+        let mut pb = bg.new_port_buffer();
+
+        let fb = ffm_vec(
+            vec![HashAndValueAndSeq {
+                hash: 1,
+                value: 1.0,
+                contra_field_index: 0,
+            }],
+            1,
+        ); // saying we have 1 field isn't entirely correct
+        assert_epsilon!(spredict2(&mut bg, &fb, &mut pb, true, true), 0.5);
+        assert_ne!(spredict2(&mut bg, &fb, &mut pb, true, true), 0.5);
+
+        // With two fields, things start to happen
+        // Since fields depend on initial randomization, these tests are ... peculiar.
```

Comment:
This is exactly why you need seeds :)

Reply:
don't you think it's okay to test that the Montecarlo prediction is not exactly equal to the single prediction but close to it?